src/test/scala/vsys/blockchain/state/contract/swap/AtomicSwapContractInvalidDiffTest.scala

```diff
@@ -0,0 +1,166 @@
+        blockDiff.txsDiff.txStatus shouldBe TransactionStatus.ContractMapValueInsufficient
+      }
+    }
+  }
+
+  val preconditionsAndAtomicSwapExpiredWithdrawInvalidTest: Gen[(GenesisTransaction, GenesisTransaction, RegisterContractTransaction,
+    ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, Long, Long)] = for {
+    (genesis, genesis2, master, user, regContract, depositVSYS, ts, fee, description, attach) <- createAndDepositVSYSAtomicSwapContractGen(1000L)
+    lockData = Seq(Longs.toByteArray(1000L), user.toAddress.bytes.arr, Sha256(Longs.toByteArray(123L)), Longs.toByteArray(ts + 100))
+    lockType = Seq(DataType.Amount, DataType.Address, DataType.ShortBytes, DataType.Timestamp)
+    lock <- lockAtomicSwapContractDataStackGen(master, regContract.contractId, lockData, lockType, attach, fee, ts)
+    withdrawData = Seq(lock.id.arr)
+    withdrawType = Seq(DataType.ShortBytes)
+    withdraw <- expireWithdrawAtomicSwapContractDataStackGen(master, regContract.contractId, withdrawData, withdrawType, attach, fee, ts)
+  } yield (genesis, genesis2, regContract, depositVSYS, lock, withdraw, ts, fee)
+
+  property("unable to withdraw before lock expires") {
+    forAll(preconditionsAndAtomicSwapExpiredWithdrawInvalidTest) { case (genesis: GenesisTransaction, genesis2: GenesisTransaction, reg: RegisterContractTransaction,
+    deposit: ExecuteContractFunctionTransaction, lock: ExecuteContractFunctionTransaction, withdraw: ExecuteContractFunctionTransaction, ts: Long, fee: Long) =>
+      assertDiffAndStateCorrectBlockTime(Seq(TestBlock.create(genesis.timestamp, Seq(genesis, genesis2)), TestBlock.create(deposit.timestamp, Seq(reg, deposit, lock))),
+        TestBlock.createWithTxStatus(lock.timestamp + 99, Seq(withdraw), TransactionStatus.Failed)) { (blockDiff, newState) =>
+        blockDiff.txsDiff.contractDB.isEmpty shouldBe true
+        blockDiff.txsDiff.contractNumDB.isEmpty shouldBe true
+        blockDiff.txsDiff.portfolios.isEmpty shouldBe false
+        blockDiff.txsDiff.txStatus shouldBe TransactionStatus.Failed
+      }
+    }
+  }
+
+  val preconditionsAndAtomicSwapLockWrongAddressInvalidTest: Gen[(GenesisTransaction, GenesisTransaction, RegisterContractTransaction,
+    ExecuteContractFunctionTransaction, ExecuteContractFunctionTransaction, Long, Long)] = for {
+    (genesis, genesis2, master, user, regContract, depositVSYS, ts, fee, description, attach) <- createAndDepositVSYSAtomicSwapContractGen(1000L)
+    lockData = Seq(Longs.toByteArray(1000L), user.toAddress.bytes.arr, Sha256(Longs.toByteArray(123L)), Longs.toByteArray(ts + 100))
+    lockType = Seq(DataType.Amount, DataType.Address, DataType.ShortBytes, DataType.Timestamp)
+    lock <- lockAtomicSwapContractDataStackGen(user, regContract.contractId, lockData, lockType, attach, fee, ts + 3)
+  } yield (genesis, genesis2, regContract, depositVSYS, lock, ts, fee)
+
+  property("wrong address cannot lock contract") {
```

Comment:
wrong address?  or insufficient balance?

Reply:
yes this should be called insufficient balance